site/src/scaffold/scripts/theming.ts

```diff
@@ -0,0 +1,147 @@
+import { parseQueryString } from './query-string';
+import { atlasTokens } from './tokens';
+
+/**
+ * The globally available descriptor of the current theme. It is updated each time setTheme is called.
+ */
+export let currentTheme: ThemeType = 'light';
+
+// * Types
+
+export type ThemeType = keyof ThemeTypeMap;
+
+/**
+ * Simple string name for each theme.
+ */
+export const themes = Object.keys(atlasTokens.themes.tokens.$themes);
+/**
+ * Data structure for easy access to theme names and css classes.
+ */
+export const themeInfo = themes.reduce((themeInfo, themeName) => {
+	themeInfo[themeName] = {
+		documentClass: `theme-${themeName}`,
+		name: themeName
+	};
+	return themeInfo;
+}, {} as ThemeTypeMap);
+/**
+ * CSS classes related to theming.
+ * Used to remove all classes before adding them next one.
+ */
+export const cssClasses = themes.map(theme => themeInfo[theme].documentClass);
+export interface ThemeTypeInfo {
+	documentClass: string;
+	name: string;
+}
+
+export interface ThemeTypeMap {
+	light: ThemeTypeInfo;
+	dark: ThemeTypeInfo;
+	'high-contrast': ThemeTypeInfo;
+}
+
+/**
+ * Using the current theme key, get the next one in the cycle.
+ * @param current The current theme's name.
+ * @returns The next theme's name.
+ */
+export function cycleKeys(current: keyof ThemeTypeMap): keyof ThemeTypeMap {
+	const i = themes.findIndex(item => item === current) || 0;
+	const next = i < 2 ? i + 1 : 0;
+	return themes[next] as keyof ThemeTypeMap;
+}
+
+/**
+ * Add class to apply the next theme, and remove previously applied classes.
+ * @param appliedTheme The name of the theme we want to apply.
+ */
+export function setThemeClass(appliedTheme: ThemeType) {
+	const htmlClassList = document.documentElement.classList;
+	for (const docClass of cssClasses) {
+		htmlClassList.remove(docClass);
+	}
+
+	htmlClassList.add(themeInfo[appliedTheme].documentClass);
+}
+
+export function setGlobalThemeValue(appliedTheme: ThemeType) {
+	return (currentTheme = appliedTheme);
+}
+
+export function storeTheme(theme: ThemeType) {
+	localStorage.setItem('theme', theme);
+}
+
+export function getPreferredTheme(prefersDarkTheme: boolean = false): ThemeType {
+	// check for user selection
+	const theme = localStorage.getItem('theme');
+	if (/^light|dark|high-contrast$/.test(theme)) {
+		return theme as ThemeType;
+	}
+
+	// check for system dark preference
+	if (prefersDarkTheme) {
+		return 'dark';
+	}
+
+	// default to light
+	return 'light';
+}
+
+/**
+ * SetTheme is the public way of changing the visual theme. It ensures we use the correct EventBus instance, while allow the user of the function to remain free of such implementation detail..
+ *
+ * @param theme The ThemeType to apply to the current window.
+ */
+export function setTheme(appliedTheme: ThemeType) {
+	const previousTheme = currentTheme;
+	setGlobalThemeValue(appliedTheme);
+	setThemeClass(appliedTheme);
+	if (previousTheme === appliedTheme) {
```

Comment:
Should this condition go right after line 99? I'm thinking that if the appliedTheme is the same as previousTheme, then there wouldn't be a need to set the current theme value and class again.

Reply:
whoops! Yes this should be changed

Reply:
Updated per your suggestion